### PR TITLE
mastodon: 3.5.0 -> 3.5.1

### PR DESCRIPTION
**Pull request by techknowlogick**
###### Description of changes

https://github.com/mastodon/mastodon/releases/tag/v3.5.1

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/mastodon/default.nix [hunk 1]
```diff
@@ -22,14 +22,6 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-Ngfs15YKLfSBOKju3BzpZFnenB370jId2G1g9Qy1y5w=";
   };
 
-  patches = [
```
Comment (erictapen) [inline pkgs/servers/mastodon/default.nix:25]: Can you remove the `fetchpatch` argument above as well?

Comment (techknowlogick) [reply pkgs/servers/mastodon/default.nix:25]: yup, thanks for the review :)

File: pkgs/servers/mastodon/default.nix [hunk 2]
```diff
@@ -22,14 +21,6 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-Ngfs15YKLfSBOKju3BzpZFnenB370jId2G1g9Qy1y5w=";
```
Comment (Izorkin) [inline pkgs/servers/mastodon/default.nix:21]: This also needs to be updated.
https://github.com/NixOS/nixpkgs/pull/167946/commits/08e700c531fd31b3a96e61a31e7b0c3af5ff3ad2 - Can you add this commit?

**Comment by erictapen** [comment]
@GrahamcOfBorg test mastodon

**Review by erictapen — APPROVED** [review]
Test runs fine for me :+1:

**Comment by Izorkin** [comment]
Is it possible to add in script update yarnOfflineCache?
How to run update.sh via nix-shell?

**Comment by erictapen** [comment]
> Is it possible to add in script update yarnOfflineCache?

I'm not sure, there would need to be a tool that can build a generic fixed-output derivation and output the new hash. Actually I'd like to limit the scope of update.sh in the future, as I find it cumbersome to have so much bash code lying around. Ideally it would just be a call to bundix to generate the lock file and everything would need to be done by hand or by another tool like nix-update.

> How to run update.sh via nix-shell?

Haven't used stable Nix for a while, but I think it would be `nix-shell -p mastodon.updateScript`? I'm running `nix shell .#mastodon.updateScript` with Nix with flakes support.

**Comment by erictapen** [comment]
@Izorkin Do you think this is ready? I'll merge it then.

**Review by Izorkin — APPROVED** [review]

**Comment by Izorkin** [comment]
> @Izorkin Do you think this is ready? I'll merge it then.

Already updated my instance - it works.

**Pull request merged**